### PR TITLE
Test/flow monitoring stats

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy [hunk 1]
```diff
@@ -345,6 +345,69 @@ class Server42RttSpec extends HealthCheckSpecification {
         revertToOrigin([flow], flowRttFeatureStartState, initialSwitchRtt)
     }
 
+    @Tidy
+    def "Rtt statistic is available for a flow in case switch is not connected to server42"() {
+        given: "Two active switches, src has server42 connected"
+        def server42SwIds = topology.getActiveServer42Switches()*.dpId
+        def switchPair = topologyHelper.switchPairs.find {
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy:352]: May be useful to search the reversed pairs as well

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy [hunk 2]
```diff
@@ -345,6 +345,69 @@ class Server42RttSpec extends HealthCheckSpecification {
         revertToOrigin([flow], flowRttFeatureStartState, initialSwitchRtt)
     }
 
+    @Tidy
+    def "Rtt statistic is available for a flow in case switch is not connected to server42"() {
+        given: "Two active switches, src has server42 connected"
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy:350]: _only_ src

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy [hunk 3]
```diff
@@ -345,6 +345,69 @@ class Server42RttSpec extends HealthCheckSpecification {
...
+        Wrappers.wait(STATS_FROM_SERVER42_LOGGING_TIMEOUT, 1) {
+            !otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
+                    [flowid   : flow.flowId,
+                     direction: "forward"]).dps.isEmpty()
+            otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
+                    [flowid   : flow.flowId,
+                     direction: "reverse"]).dps.isEmpty()
+        }
+
+        and: "Stats from flow monitoring feature for reverse direction only are available"
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy:380]: I don't quite understand. You request the same `flow.rtt` metric 4 times, but call it in different ways: 'flow monitoring stats' and 's42 stats' as completely separate entities.
So when s42 stats are not available we add the 'calculated: true' tag and call it flow monitoring stats? If yes, then test should be rephrased to highlight this more clearly. Its the same 'stat' but calculated using different tools for forward/reverse. Moreover, the only difference is the tag, so it looks like there's no need to query otsdb 4 times, you already have all the data after first 2 calls.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy [hunk 4]
```diff
@@ -345,6 +345,69 @@ class Server42RttSpec extends HealthCheckSpecification {
...
+             direction: "forward",
+             calculated: true]).dps
+            assert flMonitoringReverseData.size() >= 1
+            assert flMonitoringForwardData.isEmpty()
+        }
+
+        when: "Disable server42FlowRtt on the src switch"
+        changeFlowRttSwitch(switchPair.src, false)
+
+        then: "Stats from flow monitoring feature for forward direction are available"
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy:397]: You say 'forward' but query for 'reverse'

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy [hunk 5]
```diff
@@ -345,6 +345,69 @@ class Server42RttSpec extends HealthCheckSpecification {
...
+        def initialSrcSwS42Props = northbound.getSwitchProperties(switchPair.src.dpId).server42FlowRtt
+        changeFlowRttSwitch(switchPair.src, true)
+
+        and: "Create a flow"
+        def flowCreateTime = new Date()
+        //take shorter flowid due to https://github.com/telstra/open-kilda/issues/3720
+        def flow = flowHelperV2.randomFlow(switchPair).tap { it.flowId = it.flowId.take(25) }
+        flowHelperV2.addFlow(flow)
+
+        then: "Stats from server42 for forward direction are available"
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy:370]: only for forward

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy (hunk 4 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy:397]: thx, fixed

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy]: added

**Review by rtretyak — APPROVED** [review]

**Pull request merged**